### ISSUE TITLE
[axe-linter] Extension not working with Vue3 + Typescript in Vscode

### ISSUE THREAD

**Issue by glancel** [dequelabs/axe-core#4338]
### Product

axe Linter

### Question

Hi everyone,

The Axe linter extension for VSCode does not function properly when TypeScript is used within a Vue file. To illustrate, consider the following simple component:

```vue
<script setup lang="ts">
	defineProps({
		titles: {
			type: Array as PropType<string[]>,
			required: true,
		},
	})
</script>
<template>
	<div class="test">
		<img src="" />
		<h1>{{ titles[0] }}</h1>
	</div>
</template>
```

When attempting to insert an invalid tag, such as `<img src="" />`, no error is triggered.

However, if you remove the TypeScript annotation from titles:

```vue
<script setup lang="ts">
	defineProps({
		titles: {
			type: Array,
			required: true,
		},
	})
</script>
<template>
	<div class="test">
		<img src="" />
		<h1>{{ titles[0] }}</h1>
	</div>
</template>
```

In this scenario, everything works as expected.

Has anyone else experienced the same issue?

Thanks!!

**Comment by michael-siek** [comment]
Hi @glancel I am taking a look at this issue and will respond when possible!

**Comment by glancel** [comment]
Hi @michael-siek, is this closed because it is not an issue or will it be fixed? Thanks!!

**Comment by michael-siek** [comment]
Hey @glancel I apologize for this closing without an explanation. We currently have a fix in place in axe-linter vscode and will have an update to the extension that should fix this moving forwards. Once the fix has been released I will update this ticket accordingly.

**Comment by padmavemulapati** [comment]
Validated with the latest QA-build(v4.8.1-9047d43), now we are able to allow the ussers to lint `.vue` file with `typescript`

`.vue` test file:
```
<template>
  <div class="AriaAllowedRoleFailed">
    <input type='image' id='fail-input-image-invalid-role' role='doc-afterword' />
     <!-- Bad: Labeled using incorrectly spelled aria-labeledby -->
  </div>
</template>

<script setup lang="ts">
	defineProps({
		titles: {
			type: Array as PropType<string[]>,
			required: true,
		},
	})
</script>
```
Output:
## Screenshot:
<img width="898" alt="image" src="https://github.com/dequelabs/axe-linter/assets/47409579/090d70b5-13f2-420d-b7e5-2b68bf4dee71">


## Recording:

https://github.com/dequelabs/axe-linter/assets/47409579/26ff5a65-96b4-448a-93d9-e38f9a161382

Reproduced with the Production build:

## Recoding:

https://github.com/dequelabs/axe-linter/assets/47409579/ab0ba8f4-4033-4795-a82c-c62d96c1fac8


Using `axe-linter-connector`

Run the command:
```
AXE_SERVICE_URL=https://axe.dequelabs.com ./axe-linter-connector-macos -s ./rules/aria-allowed-attr/aria-allowed-role_ts.vue  -d .  --url=https://axe-linter.dequelabs.com --api-key=<Dev_api-key>
```

## Recording:

https://github.com/dequelabs/axe-linter/assets/47409579/ffc2008e-8485-4640-b0e4-03a4ca44498e

## Environment:
| **Label**                         | **Value** |
| -------------------------- | ------ |
| **Product**                | _axe-devtools/linter & Connector_ |
|**Version** | _v4.8.1-9047d43_ |
| **Type**           | _VSCode Bundle and Connector_ |
| **OS-Details**      | _VSCode Editor and CLI _  |

**Comment by michael-siek** [comment]
Hey @glancel just wanted to provide an update but we have just released [4.8.2 of axe-linter vscode](https://marketplace.visualstudio.com/items?itemName=deque-systems.vscode-axe-linter) which should fix this issue. Please feel free to open a new ticket or reach out in the [axe community slack channel](https://accessibility.deque.com/axe-community) if you have additional issues/questions

**Comment by padmavemulapati** [comment]
Validated with the latest QA-build(v4.8.1-9047d43), now we are able to allow the ussers to lint `.vue` file with `typescript`

`.vue` test file:
```
<template>
  <div class="AriaAllowedRoleFailed">
    <input type='image' id='fail-input-image-invalid-role' role='doc-afterword' />
     <!-- Bad: Labeled using incorrectly spelled aria-labeledby -->
  </div>
</template>

<script setup lang="ts">
	defineProps({
		titles: {
			type: Array as PropType<string[]>,
			required: true,
		},
	})
</script>
```
Output:
## Screenshot:
<img width="898" alt="image" src="https://github.com/dequelabs/axe-linter/assets/47409579/090d70b5-13f2-420d-b7e5-2b68bf4dee71">


## Recording:

https://github.com/dequelabs/axe-linter/assets/47409579/26ff5a65-96b4-448a-93d9-e38f9a161382

Reproduced with the Production build:

## Recoding:

https://github.com/dequelabs/axe-linter/assets/47409579/ab0ba8f4-4033-4795-a82c-c62d96c1fac8


Using `axe-linter-connector`

Run the command:
```
AXE_SERVICE_URL=https://axe.dequelabs.com ./axe-linter-connector-macos -s ./rules/aria-allowed-attr/aria-allowed-role_ts.vue  -d .  --url=https://axe-linter.dequelabs.com --api-key=<Dev_api-key>
```

## Recording:

https://github.com/dequelabs/axe-linter/assets/47409579/ffc2008e-8485-4640-b0e4-03a4ca44498e

With IntelliJ plugin also woriking as expected:

Seeing the linter errors

## Expected Behavior:

![image](https://github.com/dequelabs/axe-linter/assets/47409579/64fbb618-92bd-434c-8b2b-d8cc3c0a2381)



## Environment:
| **Label**                         | **Value** |
| -------------------------- | ------ |
| **Product**                | _axe-devtools/linter & Connector_ |
|**Version** | _v4.8.1-9047d43_ |
| **Type**           | _VSCode Bundle and Connector_ |
| **OS-Details**      | _VSCode Editor and CLI _  |
|**Type**|_IntelliJ-Plugin_|
|**Environment**|_IntelliJ IDEA 2023.3.4 (Ultimate Edition)-Build #IU-233.14475.28_|

**Comment by glancel** [comment]
I can confirm that the new release work as expected, thank you!!
